WordPress/Classes/ViewRelated/NUX/SignupUsernameTableViewController.swift

```diff
@@ -0,0 +1,237 @@
+        gestureRecognizer.on(call: { [weak self] (gesture) in
+            self?.view.endEditing(true)
+        })
+        gestureRecognizer.cancelsTouchesInView = false
+        view.addGestureRecognizer(gestureRecognizer)
+    }
+}
+
+// MARK: UITableViewDataSource
+
+extension SignupUsernameTableViewController {
+    fileprivate enum Sections: Int {
+        case titleAndDescription = 0
+        case searchField = 1
+        case suggestions = 2
+
+        static var count: Int {
+            return suggestions.rawValue + 1
+        }
+    }
+
+    private enum SuggestionStyles {
+        static let indentationWidth: CGFloat = 20.0
+        static let indentationLevel = 1
+    }
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return Sections.count
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch section {
+        case Sections.titleAndDescription.rawValue:
+            return 1
+        case Sections.searchField.rawValue:
+            return suggestions.count > 0 ? 1 : 0
+        case Sections.suggestions.rawValue:
+            return suggestions.count
+        default:
+            return 0
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell: UITableViewCell
+        switch indexPath.section {
+        case Sections.titleAndDescription.rawValue:
+            cell = titleAndDescriptionCell()
+        case Sections.searchField.rawValue:
+            cell = searchFieldCell()
+        case Sections.suggestions.rawValue:
+            fallthrough
+        default:
+            let suggestion = suggestions[indexPath.row]
+            cell = suggestionCell(domain: suggestion)
+        }
+        return cell
+    }
+
+    override func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
+        if section == Sections.suggestions.rawValue {
+            let footer = UIView()
+            footer.backgroundColor = WPStyleGuide.greyLighten20()
+            return footer
+        }
+        return nil
+    }
+
+    override func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        if section == Sections.suggestions.rawValue {
+            return 0.5
+        }
+        return 0
+    }
+
+    // MARK: table view cells
+
+    private func titleAndDescriptionCell() -> UITableViewCell {
+        let description = String(format: NSLocalizedString("Your username is currently \"%@\". It will be used for mentions and links, but otherwise people will just see your display name, \"%@\"", comment: "Description of how to pick a domain name during the site creation process"), currentUsername ?? "", displayName ?? "")
+        let cell = LoginSocialErrorCell(title: "", description: description)
+        cell.selectionStyle = .none
+        return cell
```

Comment:
Per the design, the usernames should be bold:
<img width="350" alt="screen shot 2018-02-25 at 11 04 26 am" src="https://user-images.githubusercontent.com/1816888/36644720-c5337f72-1a1b-11e8-9f59-993e945b5421.png">

Also, the period is missing in the implemented version. 😄